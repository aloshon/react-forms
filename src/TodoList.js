import React, {useState} from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import {v4 as uuid} from "uuid";

const TodoList = () => {
    const intitialState = []
    const [tasks, setTasks] = useState(intitialState);

    const addTask = (newTask) => {
        setTasks(tasks => [...tasks, {...newTask, id: uuid()}])
    }

    const remove = (task) => {
        setTasks(tasks.filter(t => t.id !== task))
    }

    return (
        <div>
            <h3>Todo List</h3>
            <NewTodoForm addTask={addTask}/>
            <ul className="todolist">
                {tasks.map(({id, task}) => <Todo key={id} id={id} task={task} remove={remove}/>)}
            </ul>
        </div>
    )
}

export default TodoList;