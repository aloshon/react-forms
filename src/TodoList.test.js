import React from "react";
import {render, fireEvent} from "@testing-library/react";
import TodoList from "./TodoList";

it("renders successfully", function (){
    render(<TodoList/>);
});

it("matches snapshot", function (){
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new item", function (){
    const {queryByText, getByLabelText} = render(<TodoList/>);
    const input = getByLabelText("Task");
    const btn = queryByText("Add Todo!");

    expect(queryByText('Walk Dogs')).not.toBeInTheDocument();

    fireEvent.change(input, {target: {value: 'Walk Dogs'}});
    fireEvent.click(btn);

    expect(queryByText('Walk Dogs')).toBeInTheDocument();
});