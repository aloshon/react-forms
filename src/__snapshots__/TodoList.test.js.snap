// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <form>
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        placeholder="task"
        type="text"
        value=""
      />
      <button>
        Add Todo!
      </button>
    </form>
    <ul
      class="todolist"
    />
  </div>
</DocumentFragment>
`;
